lib/puppet/provider/runlevel/systemd.rb

```diff
@@ -11,11 +11,19 @@ def level
   end
 
   def level=(should)
-    return execute([command(:systemctl),'isolate',init2systemd(@resource[:name])])
+    require 'timeout'
+
+    begin
+      Timeout::timeout(@resource[:transition_timeout]) do
+        execute([command(:systemctl),'isolate',init2systemd(@resource[:name])])
+      end
+    rescue Timeout::Error
+      raise(Puppet::Error, "Could not transition to runlevel #{@resource[:name]} within #{@resource[:transition_timeout]} seconds")
```

Comment:
How do you know the command is terminated with this raise?

Reply:
:arrow_up: This is a good question.  

Shouldn't we explicitly test for/kill it in this `rescue` block before raising the `Puppet::Error`?